projects/client/RabbitMQ.Client/src/client/impl/AutorecoveringConnection.cs

```diff
@@ -551,12 +551,8 @@ private void Init(IFrameHandler fh)
         {
             m_delegate = new Connection(m_factory, false,
                 fh, this.ClientProvidedName);
-            
-            m_recoveryThread = new Thread(MainRecoveryLoop)
-            {
-                IsBackground = true
-            };
-            m_recoveryThread.Start();
+
+            m_recoveryTask = Task.Run(MainRecoveryLoop);
```

Comment:
Use Task instead of new Thread. Utilizes the ThreadPool and gets rid of the overhead to create a new thread, also enabled turning the MainRecoveryLoop method into an async method.

Reply:
The overhead for this method is unlikely to be significant enough in practice to matter but fair enough :)

Reply:
Agreed, but using Task.Run also enables the MainRecoveryLoop to be defined as "async Task" instead of being forced to be a synchronous method :) Also delegates to using the ThreadPool for better core utilization, hopefully resulting in fewer thread blocks/stalls.